pkg/performanceprofile/controller/performanceprofile/hypershift/status/status.go

```diff
@@ -0,0 +1,123 @@
+)
+
+const (
+	ppStatusConfigMapPrefix = "status"
+)
+
+func getStatusConfigMapName(instanceName string) string {
+	return fmt.Sprintf("%s-%s", ppStatusConfigMapPrefix, instanceName)
+}
+
+func createOrUpdateStatusConfigMap(ctx context.Context, cli client.Client, cm *corev1.ConfigMap, profileName string, conditions []conditionsv1.Condition) error {
+	prevStatus, prevStatusFound, err := getPreviousStatusFrom(ctx, cli, &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cm.Name,
+			Namespace: cm.Namespace,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	npName := cm.Labels[hypershiftconsts.NodePoolNameLabel]
+	updatedStatus := status.CalculateUpdatedStatus(prevStatus, profileName, npName, conditions)
+	if updatedStatus == nil {
+		return nil
+	}
+	encodedStatus, encodeErr := yaml.Marshal(updatedStatus)
+	if encodeErr != nil {
+		return encodeErr
+	}
+	cm.Data = map[string]string{
+		hypershiftconsts.PerformanceProfileStatusKey: string(encodedStatus),
+	}
+
+	key := client.ObjectKeyFromObject(cm)
+	if prevStatusFound {
+		klog.InfoS("Updating status", "ConfigMap", key.String())
+		if err := cli.Update(ctx, cm); err != nil {
+			return fmt.Errorf("failed to update status ConfigMap %q: %w", key.String(), err)
+		}
+		klog.InfoS("Status updated ", "ConfigMap", key.String())
+	} else {
+		klog.InfoS("Creating status", "ConfigMap", key.String())
+		if err := cli.Create(ctx, cm); err != nil {
+			return fmt.Errorf("failed to create ConfigMap %q: %w", key.String(), err)
+		}
+		klog.InfoS("Status created", "ConfigMap", key.String())
+	}
+	return nil
+}
+
+func ppStatusConfigMap(instance *corev1.ConfigMap, profileName string, scheme *runtime.Scheme) (*corev1.ConfigMap, error) {
+	nodePoolNamespacedName, ok := instance.Annotations[hypershiftconsts.NodePoolNameLabel]
+	if !ok {
+		return nil, fmt.Errorf("annotation %q not found in ConfigMap %q annotations", hypershiftconsts.NodePoolNameLabel, client.ObjectKeyFromObject(instance).String())
+	}
+	cm := handler.ConfigMapMeta(getStatusConfigMapName(instance.Name), profileName, instance.GetNamespace(), nodePoolNamespacedName)
+	err := controllerutil.SetControllerReference(instance, cm, scheme)
+	if err != nil {
+		return nil, err
+	}
+	cm.Labels[hypershiftconsts.NTOGeneratedPerformanceProfileStatusConfigMapLabel] = "true"
+
+	return cm, nil
+}
+
+func getTunedConditions(ctx context.Context, cli client.Client) ([]conditionsv1.Condition, error) {
+	tunedProfileList := &tunedv1.ProfileList{}
+	if err := cli.List(ctx, tunedProfileList); err != nil {
+		klog.Errorf("Cannot list Tuned Profiles: %v", err)
+		return nil, err
+	}
+
+	messageString := status.GetTunedProfilesMessage(tunedProfileList.Items)
+	if len(messageString) == 0 {
+		return nil, nil
+	}
+	return status.GetDegradedConditions(status.ConditionReasonTunedDegraded, messageString), nil
+}
+
+func getPreviousStatusFrom(ctx context.Context, cli client.Client, cm *corev1.ConfigMap) (*performancev2.PerformanceProfileStatus, bool, error) {
```

Comment:
non-blocking question: is this get cached somehow somewhere (client-go perhaps) right?

Reply:
Caching when we're having two clients (i.e. two etcds) is something that deserves more intention and deeper investigation. ATM I do not have all the answers unfortunately, but what I'm knowing for sure is that we're caching only the hosted-cluster objects:
https://github.com/openshift/cluster-node-tuning-operator/blob/master/cmd/cluster-node-tuning-operator/main.go#L137

Reply:
ok, good enough for now